File: code/JL_OTA_V1.7.0_SDK_V1.9.0/otasdk/src/main/java/com/jieli/otasdk/dialog/DialogSeekBar.java
package com.jieli.otasdk.dialog;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.jieli.otasdk.databinding.FragmentDialogFileTransferBinding;
import com.jieli.otasdk.databinding.FragmentDialogSeekBarBinding;

/**
 * @ClassName: DialogSeekBar
 * @Description: 滑动条弹窗
 * @Author: ZhangHuanMing
 * @CreateDate: 2022/10/21 10:19
 */
public class DialogSeekBar extends DialogFragment {
    public DialogSeekBarListener mListener = null;
    private FragmentDialogSeekBarBinding mBinding;
    private final int MAX = 512;
    private final int MIN = 23;
    public String title = null;
    public int progress = MIN;
    public int max = MAX;
    public int min = MIN;
    public String leftText = null;
    public String rightText = null;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        //设置dialog的基本样式参数
        requireDialog().requestWindowFeature(Window.FEATURE_NO_TITLE);
        Window window = requireDialog().getWindow();
        if (window != null) {
            //去掉dialog默认的padding
            window.getDecorView().setPadding(0, 0, 0, 0);
            WindowManager.LayoutParams layoutParams = window.getAttributes();
            layoutParams.width = WindowManager.LayoutParams.WRAP_CONTENT;
            layoutParams.height = WindowManager.LayoutParams.WRAP_CONTENT;
            layoutParams.gravity = Gravity.CENTER;
            window.setAttributes(layoutParams);
            window.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        }
        mBinding = FragmentDialogSeekBarBinding.inflate(inflater, container, false);
        return mBinding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (title != null) {
            mBinding.tvDialogTitle.setVisibility(View.VISIBLE);
            mBinding.tvDialogTitle.setText(title);
        }
        mBinding.seekBar.setProgress(progress);
        mBinding.seekBar.setMax(max);
        mBinding.seekBar.setMin(min);
        mBinding.tvMax.setText(String.valueOf(max));
        mBinding.tvMin.setText(String.valueOf(min));
        if (leftText != null) {
            mBinding.btLeft.setVisibility(View.VISIBLE);
            mBinding.btLeft.setText(leftText);
        }
        if (rightText != null) {
            mBinding.btRight.setVisibility(View.VISIBLE);
            mBinding.btRight.setText(rightText);
        }
        if (rightText != null && leftText != null) {
            mBinding.view1.setVisibility(View.VISIBLE);
        }
        mBinding.btLeft.setOnClickListener(v -> {
            if (mListener != null) {
                mListener.onLeftButtonClick(mBinding.seekBar.getProgress());
            }
        });
        mBinding.btRight.setOnClickListener(v -> {
            if (mListener != null) {
                mListener.onRightButtonClick(mBinding.seekBar.getProgress());
            }
        });
    }
}

File: code/JL_OTA_Android_V1.9.0_SDK_V1.11.0/otasdk/src/main/java/com/jieli/otasdk/ui/dialog/DialogSeekBar.kt
package com.jieli.otasdk.ui.dialog

import android.graphics.Typeface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import com.jieli.jl_bt_ota.constant.BluetoothConstant
import com.jieli.otasdk.R
import com.jieli.otasdk.databinding.FragmentDialogSeekBarBinding
import com.jieli.otasdk.util.gone
import com.jieli.otasdk.util.show

/**
 * @ClassName: DialogSeekBar
 * @Description: 滑动条弹窗
 * @Author: ZhangHuanMing
 * @CreateDate: 2022/10/21 10:19
 */
class DialogSeekBar private constructor(builder: Builder) : CommonDialog(builder) {

    companion object {
        const val MAX = BluetoothConstant.BLE_MTU_MAX + 3
        const val MIN = BluetoothConstant.BLE_MTU_MIN + 3
    }

    private lateinit var binding: FragmentDialogSeekBarBinding

    override fun createView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        FragmentDialogSeekBarBinding.inflate(inflater, container, false).also {
            binding = it
            return it.root
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initUI()
    }

    fun getProgress(): Int = binding.seekBar.progress

    private fun initUI() {
        if (builder !is Builder) return
        binding.tvDialogTitle.apply {
            if (builder.titleStyle == null) gone() else show()
            builder.titleStyle?.let {
                text = it.text
                setTextColor(
                    ContextCompat.getColor(
                        context,
                        if (it.color == 0) R.color.text_color else it.color
                    )
                )
                textSize = (if (it.size == 0) 18 else it.size).toFloat()
                typeface =
                    if (it.isBold) Typeface.DEFAULT_BOLD else Typeface.defaultFromStyle(Typeface.NORMAL)
                gravity = it.gravity
                it.onClick?.let { listener ->
                    setOnClickListener { view ->
                        listener.onClick(this@DialogSeekBar, view)
                    }
                }
            }
        }
        binding.btnCancel.apply {
            if (builder.cancelBtnStyle == null) gone() else show()
            builder.cancelBtnStyle?.let {
                text = it.text.ifEmpty { getString(R.string.cancel) }
                setTextColor(
                    ContextCompat.getColor(
                        context,
                        if (it.color == 0) R.color.black_66000000 else it.color
                    )
                )
                textSize = (if (it.size == 0) 18 else it.size).toFloat()
                typeface =
                    if (it.isBold) Typeface.DEFAULT_BOLD else Typeface.defaultFromStyle(Typeface.NORMAL)
                gravity = it.gravity
                setOnClickListener { view ->
                    it.onClick?.onClick(this@DialogSeekBar, view)
                }
            }
        }
        binding.btnConfirm.apply {
            if (builder.sureBtnStyle == null) gone() else show()
            builder.sureBtnStyle?.let { sureBtn ->
                text = sureBtn.text.ifEmpty { getString(R.string.confirm) }
                setTextColor(
                    ContextCompat.getColor(
                        context,
                        if (sureBtn.color == 0) R.color.blue_558CFF else sureBtn.color
                    )
                )
                textSize = (if (sureBtn.size == 0) 18 else sureBtn.size).toFloat()
                typeface =
                    if (sureBtn.isBold) Typeface.DEFAULT_BOLD else Typeface.defaultFromStyle(
                        Typeface.NORMAL
                    )
                gravity = sureBtn.gravity
                setOnClickListener { view ->
                    sureBtn.onClick?.onClick(this@DialogSeekBar, view)
                }
            }
        }
        binding.seekBar.apply {
            builder.progressStyle.let { progressStyle ->
                max = progressStyle.max.toFloat()
                min = progressStyle.min.toFloat()
                setProgress(progressStyle.progress.toFloat())
                binding.tvMax.text = progressStyle.max.toString()
                binding.tvMin.text = progressStyle.min.toString()
            }
        }

        if (builder.cancelBtnStyle != null && builder.sureBtnStyle != null) {
            binding.viewHorizontalLine.show()
            binding.viewLine.show()
            binding.btnCancel.show()
            binding.btnConfirm.show()
        } else if (builder.cancelBtnStyle == null && builder.sureBtnStyle == null) {
            binding.viewHorizontalLine.gone()
            binding.viewLine.gone()
            binding.btnCancel.gone()
            binding.btnConfirm.gone()
        } else {
            binding.viewHorizontalLine.show()
            binding.viewLine.gone()
            if (builder.cancelBtnStyle != null) {
                binding.btnConfirm.gone()
                binding.btnCancel.show()
            }
            if (builder.sureBtnStyle != null) {
                binding.btnCancel.gone()
                binding.btnConfirm.show()
            }
        }
    }

    data class ProgressStyle(
        var progress: Int = MIN,
        var max: Int = MAX,
        var min: Int = MIN
    )

    class Builder : CommonDialog.Builder() {
        var titleStyle: TextStyle? = null
        var cancelBtnStyle: ButtonStyle? = null
        var sureBtnStyle: ButtonStyle? = null
        var progressStyle: ProgressStyle = ProgressStyle()

        init {
            cancelable = false
        }

        fun title(text: String): Builder {
            if (titleStyle == null) {
                titleStyle = TextStyle(text, isBold = true)
            } else {
                titleStyle?.text = text
            }
            return this
        }

        fun cancelBtn(
            text: String = "",
            color: Int = 0,
            isBold: Boolean = true,
            listener: OnViewClick? = null
        ): Builder {
            if (cancelBtnStyle == null) {
                cancelBtnStyle = ButtonStyle(text)
            } else {
                cancelBtnStyle?.text = text
            }
            cancelBtnStyle?.let {
                it.color = color
                it.isBold = isBold
                it.onClick = listener
            }
            return this
        }

        fun confirmBtn(
            text: String = "",
            color: Int = 0,
            isBold: Boolean = true,
            listener: OnViewClick? = null
        ): Builder {
            if (sureBtnStyle == null) {
                sureBtnStyle = ButtonStyle(text)
            } else {
                sureBtnStyle?.text = text
            }
            sureBtnStyle?.let {
                it.color = color
                it.isBold = isBold
                it.onClick = listener
            }
            return this
        }

        fun progress(progress: Int, min: Int = MIN, max: Int = MAX): Builder {
            progressStyle.let {
                it.progress = progress
                it.min = min
                it.max = max
            }
            return this
        }

        override fun build(): DialogSeekBar = DialogSeekBar(this)

    }
}
